Babylon/groups/azure/groups/arm/commands/delete.py

```diff
@@ -0,0 +1,41 @@
+import logging
+from typing import Optional
+
+from azure.core.exceptions import HttpResponseError
+from azure.mgmt.resource import ResourceManagementClient
+from click import argument
+from click import command
+from click import make_pass_decorator
+
+from ......utils.decorators import require_platform_key
+
+logger = logging.getLogger("Babylon")
+
+pass_arm_client = make_pass_decorator(ResourceManagementClient)
+
+
+@command()
+@pass_arm_client
+@argument("deployment_name")
+@require_platform_key("resource_group_name", "resource_group_name")
+def delete(
+    arm_client: ResourceManagementClient,
+    deployment_name: str,
+    resource_group_name: str,
+) -> Optional[str]:
+    """Delete a resource deployment via arm deployment."""
+
+    logger.info(f"Deleting resource deployment {deployment_name} ...")
+    try:
+        poller = arm_client.deployments.begin_delete(
+            resource_group_name=resource_group_name,
+            deployment_name=deployment_name,
+        )
+    except HttpResponseError as _e:
+        logger.error(f"An error occurred : {_e.message}")
+        return
+
+    logger.debug(poller.result())
+    logger.info(f"Deployment {deployment_name} deleted with status : {poller.status()}")
+
+    return poller.status()
```

Comment:
You may rebase to main and use the new `CommandResponse` class !
See  `azure acr get-all` for an example